### DISCUSSION TITLE
Warm Transfer

### DISCUSSION THREAD

**Discussion by clegeard** [amazon-connect/amazon-connect-streams#684] [Q&A]
Hi,
I am developing a custom CCP in an Angular project and I would like to do warm transfers using Quick Connect, but I only managed to do cold transfers. The call is transferred directly to the quick connect end point.

This is the code I'm using

```
/**
	 * Transfers to Quick Connect
	 * @param agent 
	 * @param contact 
	 */
	transferToQuickConnect(agent: connect.Agent, contact: connect.Contact): void {
		agent.getEndpoints(agent.getAllQueueARNs(), {
			success: (data) => {
				const quickConnectNumber = data.endpoints.find(endpoint => endpoint.name === queueName.transferToServiceProvider);
				if (quickConnectNumber) {
					this.addConnection(agent, contact, quickConnectNumber);
				} else {
					this.errorTransfer(this._translateService.instant('CCP.Missing Service Provider Number'));
				}

			},
			failure: (error) => {
				this.logError('failed getAllQueueARNs');
				this.errorTransfer('Missing Queues');
				this.logError(error);
			}
		});

	}

/**
	 * Adds a connection to current call
	 * @param agent 
	 * @param contact 
	 * @param endPoint 
	 */
	addConnection(agent: connect.Agent, contact: connect.Contact, endPoint?: connect.Endpoint): void {
		agent.getContacts(connect.ContactType.VOICE)[0].addConnection(endPoint, {
			success: () => {
				this.logInfo(`transferring to ${endPoint}`);
				contact.getAgentConnection().destroy({
					success: () => {
						this.logInfo(`destroying ${contact}`);
					},
					failure: (error) => {
						const errMessage = 'Initial call not destroyed';
						this.errorTransfer(errMessage);
						this.logError(errMessage);
						this.logError(error);
					}
				});
			},
			failure: (error) => {
				const errMessage = 'Agent Contacts Not Found';
				this.errorTransfer(errMessage);
				this.logError(errMessage);
				this.logError(error);
			}
		});
	}
```
I'm assuming that the call is tranferred directly because I'm destroying the first connection after adding the end point. 
I tried to not destroy the connection but it's not working either.
Could you somebody share some code on how this is done?
Thanks

**Comment by jagadeeshaby** [comment]
Could you expand more on your use cases?  not able to connect warm/cold transfer use cases back to Streams functionality.

**Reply by clegeard** [reply]
Hi,
I have managed to fix the issue with the help of the Amazon Support Team.
Thanks

While I have you around here, maybe you can help me with something I'm struggling with right now.

We currently have a custom CCP working in our website and I'm trying to display to the operators errors whenever there is problem with the softphone.
I can see the errors are thrown by the library in the browser console, but when I try to capture them using the corresponding method:
`agent.onSoftphoneError((error)` => { --- `})`
nothing happens.

The errors thrown by the connect streams library are for example:

``
[2023-08-02T11:31:29.992Z] [ERROR] [a6009786-2053-4b22-abfa-56aedd921de6]:  Softphone error occurred :  microphone_not_shared Your microphone is not enabled in your browser. connect-streams-min.518cb193.js:1:436319
[2023-08-02T11:31:30.024Z] [ERROR] [a6009786-2053-4b22-abfa-56aedd921de6]: 'agent::softphone_error' event handler failed. connect-streams-min.518cb193.js:1:436319
``

I assumed we should be able to capture the errors using agent.onSoftphoneError, but it seems it is not the case.

What am I doing wrong?

Thanks
